tasks/git-advanced.md

```diff
@@ -0,0 +1,140 @@
+
+Basically, you have successfully completed this course
+[entry task](git-intro.md) and
+[git collaboration task](git-collaboration.md)
+
+## Topics
+
+1. Commit often. How often and why?
+1. Branching for experiments and easy roll-back. Fail no fear!
+1. Oh shit, git! Undoing things.
+1. `merge` vs `rebase`. Which is when.
+1. Squash 'em all!
+1. Never, ever, ever rewrite history! Or?
+1. Conflict resolution. Peace!
+1. Collaboration work flows
+1. Issue Driven Development
+
+Some topics above are really short, others take time
+for they are better comprehended when practising.
+
+## Materials
+
+ * At [Learn git branching](https://learngitbranching.js.org/)
+   refresh your skills:
+   - Main: Moving Work Around, Advanced Topics
+   - Remote: all exercises
+ * git flows
+   - [A successful Git branching model @nvie](https://nvie.com/posts/a-successful-git-branching-model/)
+   - [Issues with "nvie" git-flow](http://scottchacon.com/2011/08/31/github-flow.html)
+   - [Understanding the GitHub flow](https://guides.github.com/introduction/flow/)
+   - [Introduction to GitLab Flow](https://docs.gitlab.com/ee/workflow/gitlab_flow.html)
+   - [git-flow vs github-flow vs gitlab-flow](https://stackoverflow.com/questions/18188492/what-are-the-pros-and-cons-of-git-flow-vs-github-flow/35915110#35915110)
+   - [OneFlow - a Git branching model and workflow](https://www.endoflineblog.com/oneflow-a-git-branching-model-and-workflow)
+   - [Comparing Workflows @Atlassian](https://www.atlassian.com/git/tutorials/comparing-workflows)
+   - [DM Development Workflow with Git, GitHub, JIRA and Jenkins](https://developer.lsst.io/processes/workflow.html)
+   - [Git Workflows for Pros: A Good Git Guide](https://www.toptal.com/git/git-workflows-for-pros-a-good-git-guide)
+ * git automation tools:
+   - [hub from github](https://hub.github.com/)
+   - [gitflow-avh](https://github.com/petervanderdoes/gitflow-avh)
+   - [git-town](https://github.com/Originate/git-town)
+   - [lazygit](https://github.com/jesseduffield/lazygit)
+   - [git aliases](https://git-scm.com/book/en/v2/Git-Basics-Git-Aliases)
+   - [gitalias](https://github.com/GitAlias/gitalias)
+ * [Advanced Git Tricks (video)](https://vimeo.com/49444883)
+ * [Git Patterns and Anti-Patterns](https://dzone.com/refcardz/git-patterns-and-anti-patterns?chapter=1)
+ * [Oh shit, git!](http://ohshitgit.com/)
+ * [Flight rules for git](https://github.com/k88hudson/git-flight-rules)
+ * [GitHub Learning Lab](https://lab.github.com/courses)
+ * [octotree](https://github.com/ovity/octotree) // [Chrome github navigation plug-in](https://chrome.google.com/webstore/detail/octotree/bkhaagjahfmjljalopjnoealnfndnagc?hl=en-US)
+ * [Скринкаст по git (videos)](https://learn.javascript.ru/screencast/git)
+ * [_placeholder for workshop videocast_](#)
+
+---
+
+## Side notes
+
+You don't need to read the notes below. These
+are rather a collection of possible theses.
+
+<details><summary>TL;DR</summary>
+<p>
```

Comment:
Browsers automatically put `</p>` if next element is `ul` as like as in your case.

Reply:
Thanks for a good note.